### PR TITLE
New Resource : `azurerm_arc_resource_bridge_appliance`

### PR DESCRIPTION
**Pull request by sinbai**
Swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resourceconnector/resource-manager/Microsoft.ResourceConnector/stable/2022-10-27/appliances.json

Test results:
```
PASS: TestAccArcResourceBridgeAppliance_basic (221.73s)
PASS: TestAccArcResourceBridgeAppliance_update (343.70s)
PASS: TestAccArcResourceBridgeAppliance_complete (243.17s)
PASS: TestAccArcResourceBridgeAppliance_requiresImport (218.86s)
```
Note: Need to run `az provider register --namespace Microsoft.ResourceConnector` to register provider before running tests.

### CODE REVIEW

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @sinbai 

Thanks for this PR - I've taken a look through and left some comments inline, if those can be addressed then we should be able to take another look here.

It's worth noting that when a field has to be set via the Update method, that we can first Create the resource and then use the Update API to update this value - as such we don't need to ask users to conditionally apply their Terraform Configuration in two stages.

It's also worth noting there's conflation between the Appliance and the Resource Connector Appliance in the documentation, presumably the Appliance has to exist within AKS/VMware etc first? If so, then we're also missing test cases showing this working end-to-end - could we clarify the behaviour of this service to make it fully usable here?

Thanks!

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			if model.PublicKey != "" {
+				return fmt.Errorf("the public key can not be set when creating %s. it could be set after a deploy as an update operation", id)
+			}
+
+			if model.Version != "" {
+				return fmt.Errorf("the version cannot be set when creating %s. it could be set from upgrade call", id)
+			}
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: For `PublicKey` property, I fixed it as the suggested. For the `Version` property, I have removed it as it is not a user settable property after confirming with service team.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+			parameters := appliances.Appliance{
+				Location: model.Location,
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+			if parameters == nil {
+				return fmt.Errorf("retrieving %s: model was nil", *id)
+			}
+
+			identity, err := identity.ExpandSystemAssignedFromModel(model.Identity)
+			if err != nil {
+				return fmt.Errorf("expanding SystemAssigned Identity: %+v", err)
+			}
+
+			parameters.Identity = identity
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+				parameters.Properties = &appliances.ApplianceProperties{}
+			}
+
+			if metadata.ResourceData.HasChange("public_key") {
+				parameters.Properties.PublicKey = pointer.To(model.PublicKey)
+			}
+
+			if metadata.ResourceData.HasChange("version") {
+				parameters.Properties.Version = pointer.To(model.Version)
+			}
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+				},
+				Tags: pointer.To(model.Tags),
+			}
+
+			identity, err := identity.ExpandSystemAssignedFromModel(model.Identity)
+			if err != nil {
+				return fmt.Errorf("expanding SystemAssigned Identity: %+v", err)
+			}
+
+			parameters.Identity = identity
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+
+			state.Tags = pointer.From(resp.Model.Tags)
+			if v := resp.Model.Properties; v != nil {
+				state.Distro = pointer.From(v.Distro)
+				state.PublicKey = pointer.From(v.PublicKey)
+				state.Version = pointer.From(v.Version)
+				if p := v.InfrastructureConfig; p != nil {
+					state.Provider = pointer.From(p.Provider)
+				}
+			}
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+func (r ResourceConnectorApplianceResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			rd := metadata.ResourceDiff
+			if rd.HasChange("public_key") {
+				old, _ := rd.GetChange("public_key")
+				if old.(string) != "" {
+					return fmt.Errorf("the public_key can not be updated once it is set")
+				}
+			}
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource_test.go [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+				check.That(data.ResourceName).ExistsInAzure(l),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccResourceConnectorAppliance_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_resource_connector_appliance", "test")
+	l := ResourceConnectorApplianceResource{}
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource_test.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource_test.go [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ResourceConnectorApplianceResource struct{}
+
+func TestAccResourceConnectorAppliance_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_resource_connector_appliance", "test")
+	l := ResourceConnectorApplianceResource{}
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource_test.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource_test.go [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+				check.That(data.ResourceName).ExistsInAzure(l),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccResourceConnectorAppliance_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_resource_connector_appliance", "test")
+	l := ResourceConnectorApplianceResource{}
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource_test.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource_test.go [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+				check.That(data.ResourceName).ExistsInAzure(l),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccResourceConnectorAppliance_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_resource_connector_appliance", "test")
+	l := ResourceConnectorApplianceResource{}
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource_test.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource_test.go [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r ResourceConnectorApplianceResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-Appliances-%[1]d"
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource_test.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource_test.go [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+  location = "%[2]s"
+}
+resource "azurerm_resource_connector_appliance" "test" {
+  name                    = "acctestAppliances-%[1]d"
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource_test.go]: Fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+page_title: "Azure Resource Manager: azurerm_resource_connector_appliance"
+description: |-
+  Manages an Appliance.
+---
+
+# azurerm_resource_connector_appliance
+
+Manages an Appliances.
+
+-> **Note:** Before using this resource, it's required to submit the request of registering the providers with Azure CLI `az provider register --namespace Microsoft.ResourceConnector`.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 15]
```diff
@@ -0,0 +1,92 @@
+---
+subcategory: "Resource Connector"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_resource_connector_appliance"
+description: |-
+  Manages an Appliance.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Just the connection. The description has been updated.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+---
+
+# azurerm_resource_connector_appliance
+
+Manages an Appliances.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Just the connection. It has been updated.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+
+-> **Note:** Before using this resource, it's required to submit the request of registering the providers with Azure CLI `az provider register --namespace Microsoft.ResourceConnector`.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example"
+  location = "eastus"
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+
+The following arguments are supported:
+
+* `name` - (Required) The Name which should be used for this Appliance. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) Specifies the resource group where the resource exists. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure Region where the Appliance should exist. Changing this forces a new resource to be created.
+
+* `distro` - (Required) Specifies the resource group where the resource exists. The possible value is `AKSEdge`.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+    "hello" = "world"
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The Name which should be used for this Appliance. Changing this forces a new resource to be created.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+
+* `location` - (Required) The Azure Region where the Appliance should exist. Changing this forces a new resource to be created.
+
+* `distro` - (Required) Specifies the resource group where the resource exists. The possible value is `AKSEdge`.
+
+* `identity` - (Required) An `identity` block as defined below. Changing this forces a new resource to be created.
+
+* `infrastructure_provider`- (Required) The infrastructure provider about the connected Appliance. Possible values are `HCI`,`SCVMM` and `VMWare`. Changing this forces a new resource to be created.
+
+* `public_key`- (Required) The `public_key` is an RSA public key in PKCS1 format encoded in base64.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The Name which should be used for this Appliance. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) Specifies the resource group where the resource exists. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure Region where the Appliance should exist. Changing this forces a new resource to be created.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Resource Connector Appliance, fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The Name which should be used for this Appliance. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) Specifies the resource group where the resource exists. Changing this forces a new resource to be created.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+
+* `distro` - (Required) Specifies the resource group where the resource exists. The possible value is `AKSEdge`.
+
+* `identity` - (Required) An `identity` block as defined below. Changing this forces a new resource to be created.
+
+* `infrastructure_provider`- (Required) The infrastructure provider about the connected Appliance. Possible values are `HCI`,`SCVMM` and `VMWare`. Changing this forces a new resource to be created.
+
+* `public_key`- (Required) The `public_key` is an RSA public key in PKCS1 format encoded in base64.
+
+* `tags` - (Optional) A mapping of tags which should be assigned to the Appliance.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Resource Connector Appliance, fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 24]
```diff
@@ -0,0 +1,302 @@
...
+		},
+
+		"tags": commonschema.Tags(),
+
+		"version": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"latest",
+			}, false),
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: After confirming with service team, the `version` is not a user settable property, I have removed it.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+
+* `version` - (Optional) The version of the Appliance. Possible values is `latest`.
+
+-> **NOTE:** The `version` cannot be populated apart from upgrade call.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: After confirming with service team, the version is not a user settable property, I have removed it.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 26]
```diff
@@ -0,0 +1,92 @@
...
+* `read` - (Defaults to 5 minutes) Used when retrieving the Appliance.
+* `update` - (Defaults to 30 minutes) Used when updating the Appliance.
+* `delete` - (Defaults to 30 minutes) Used when deleting the Appliance.
+
+## Import
+
+Appliance can be imported using the `resource id`, e.g.
+
+```shell
+terraform import azurerm_resource_connector_appliance.example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/providers/Microsoft.ResourceConnector/apliances/appliancesExample
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: appliances, fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 27]
```diff
@@ -0,0 +1,92 @@
...
+* `id` - The ID of the Appliance.
+
+## Timeouts
+
+The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/language/resources/syntax#operation-timeouts) for certain actions:
+
+* `create` - (Defaults to 60 minutes) Used when creating the Appliance.
+* `read` - (Defaults to 5 minutes) Used when retrieving the Appliance.
+* `update` - (Defaults to 30 minutes) Used when updating the Appliance.
+* `delete` - (Defaults to 30 minutes) Used when deleting the Appliance.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Resource Connector Appliance, fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 28]
```diff
@@ -0,0 +1,92 @@
...
+An `identity` block supports the following:
+
+* `type` - (Required) Specifies the type of Managed Service Identity that should be configured on this Appliance. The only possible value is `SystemAssigned`.
+
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the Appliance.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Resource Connector Appliance, fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 29]
```diff
@@ -0,0 +1,92 @@
...
+
+* `identity` - (Required) An `identity` block as defined below. Changing this forces a new resource to be created.
+
+* `infrastructure_provider`- (Required) The infrastructure provider about the connected Appliance. Possible values are `HCI`,`SCVMM` and `VMWare`. Changing this forces a new resource to be created.
+
+* `public_key`- (Required) The `public_key` is an RSA public key in PKCS1 format encoded in base64.
+
+* `tags` - (Optional) A mapping of tags which should be assigned to the Appliance.
+
+* `version` - (Optional) The version of the Appliance. Possible values is `latest`.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Resource Connector Appliance, fixed.

**Comment by sinbai** [comment]
> hi @sinbai
> 
> Thanks for this PR - I've taken a look through and left some comments inline, if those can be addressed then we should be able to take another look here.
> 
> It's worth noting that when a field has to be set via the Update method, that we can first Create the resource and then use the Update API to update this value - as such we don't need to ask users to conditionally apply their Terraform Configuration in two stages.
> 
> It's also worth noting there's conflation between the Appliance and the Resource Connector Appliance in the documentation, presumably the Appliance has to exist within AKS/VMware etc first? If so, then we're also missing test cases showing this working end-to-end - could we clarify the behaviour of this service to make it fully usable here?
> 
> Thanks!

@tombuildsstuff  thanks for your detailed feedback. I have fixed the inline feedback. Could you please take another look?

For above `test cases showing this working end-to-end`, the following is the feedback I got from the service team.
```
Appliance RP is a hybrid resource provider service. We have a few operations that deploy a cluster on prem for customers like Validate, Prepare, Deploy are run on prem only and can be done via Arc Appliance CLI as this requires specific parameters depending on the on prem stack being used to deploy the cluster on. 

We do create an Azure resource to help link the on prem resource to Azure, and that is done using  RP operations (that we support using various SDKs) like Create, Update, Get, List, etc, for which we are also looking to support Terraform as the RP is GAing.
```

> Link to public doc - [Azure Arc resource bridge (preview) overview - Azure Arc | Microsoft Learn](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fazure-arc%2Fresource-bridge%2Foverview&data=05%7C01%7Cv-elenaxin%40microsoft.com%7Cc0f9934bf9d24dab02a408dbb31b47b5%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638300702068993345%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=frjs4LxVjoCP8z76XX%2Fxs60t0oU6IuB8%2B08WC6GC5Vo%3D&reserved=0)

Per above feedback, I assume that we could not add end-to-end testing in TF, what do you think?

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for making those changes @sinbai.

There are a few things that still need to be fixed.

Also since this is from the Azure Arc product line this resource should be renamed to `azurerm_arc_bridge_appliance`/`azurerm_arc_bridge_connector_appliance` or something of the sort? This is because we should go with names that are familiar to the users, in this case since this is being marketed as Arc Bridge. Also existing arc resources within the provider like the Arc Kubernetes cluster or Arc Virtual Machine are also named as such.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 30]
```diff
@@ -0,0 +1,275 @@
...
+
+type ApplianceModel struct {
+	Name              string                         `tfschema:"name"`
+	ResourceGroupName string                         `tfschema:"resource_group_name"`
+	Location          string                         `tfschema:"location"`
+	Distro            appliances.Distro              `tfschema:"distro"`
+	Identity          []identity.ModelSystemAssigned `tfschema:"identity"`
+	Provider          appliances.Provider            `tfschema:"infrastructure_provider"`
+	PublicKey         string                         `tfschema:"public_key"`
+	Tags              map[string]string              `tfschema:"tags"`
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 31]
```diff
@@ -0,0 +1,275 @@
...
+
+			parameters := appliances.Appliance{
+				Location: location.Normalize(model.Location),
+				Properties: &appliances.ApplianceProperties{
+					Distro: pointer.To(model.Distro),
+					InfrastructureConfig: &appliances.AppliancePropertiesInfrastructureConfig{
+						Provider: pointer.To(model.Provider),
+					},
+				},
+				Tags: pointer.To(model.Tags),
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 32]
```diff
@@ -0,0 +1,275 @@
...
+				identity, err := identity.ExpandSystemAssignedFromModel(model.Identity)
+				if err != nil {
+					return err
+				}
+
+				parameters.Identity = identity
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				parameters.Tags = pointer.To(model.Tags)
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: internal/services/resourceconnector/resourceconnector_appliance_resource.go [hunk 33]
```diff
@@ -0,0 +1,275 @@
...
+
+			state := ApplianceModel{
+				Name:              id.ApplianceName,
+				ResourceGroupName: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+				state.Identity = identity.FlattenSystemAssignedToModel(model.Identity)
+				state.Tags = pointer.From(resp.Model.Tags)
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/resourceconnector/resourceconnector_appliance_resource.go]: Fixed.

File: website/docs/r/resource_connector_appliance.html.markdown [hunk 34]
```diff
@@ -0,0 +1,86 @@
...
+
+* `location` - (Required) The Azure Region where the Resource Connector Appliance should exist. Changing this forces a new resource to be created.
+
+* `distro` - (Required) Specifies a supported Fabric/Infrastructure for this Resource Connector Appliance. The possible value is `AKSEdge`.
+
+* `identity` - (Required) An `identity` block as defined below. Changing this forces a new resource to be created.
+
+* `infrastructure_provider`- (Required) The infrastructure provider about the connected Resource Connector Appliance. Possible values are `HCI`,`SCVMM` and `VMWare`. Changing this forces a new resource to be created.
+
+* `public_key`- (Optional) The `public_key` is an RSA public key in PKCS1 format encoded in base64. Changing this forces a new resource to be created.
```
[the remark this answers is not in the document]
Comment (sinbai) [reply website/docs/r/resource_connector_appliance.html.markdown]: Thanks for your explanation, fixed:)

**Comment by sinbai** [comment]
> Thanks for making those changes @sinbai.
> 
> There are a few things that still need to be fixed.
> 
> Also since this is from the Azure Arc product line this resource should be renamed to `azurerm_arc_bridge_appliance`/`azurerm_arc_bridge_connector_appliance` or something of the sort? This is because we should go with names that are familiar to the users, in this case since this is being marketed as Arc Bridge. Also existing arc resources within the provider like the Arc Kubernetes cluster or Arc Virtual Machine are also named as such.

@stephybun thanks for you detailed feedback. I updated the resource name to `azurerm_arc_resource_bridge_appliance` based on [What is Azure Arc resource bridge (preview)](https://learn.microsoft.com/en-us/azure/azure-arc/resource-bridge/overview), WDYT? Also, I updated the other comments, could you please take another look?

**Review by stephybun — APPROVED** [review]
Thanks @sinbai LGTM 🍰

**Pull request merged**